community/c++-algo/include/GameLib/src/navigation.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+Description: Implementations for the navigation header.
+Last Modified: 08 Apr 2019
+Author: Patrik Dobias
+*/
+
+#include "navigation.h"
+
+namespace terminal {
+
+    using std::numeric_limits;
+    using std::find;
+    using std::vector;
+    using std::queue;
+    using std::cerr;
+
+    /// Constructor for ShortestPathFinder.
+    ShortestPathFinder::ShortestPathFinder(const GameState& gameState) : gameState(gameState) { 
+        initialized = false;
+    }
+
+    /// Initializes the map.
+    /// @param gameState A GameState object representing the gamestate we want to find path for (by reference).
+    void ShortestPathFinder::initializeMap() {
+        initialized = true;
+        gameMap.reserve(gameState.ARENA_SIZE);
+
+        for (size_t i = 0; i < gameState.ARENA_SIZE; i++) {
+            gameMap.push_back(vector<Node>());
+            gameMap[i].reserve(gameState.ARENA_SIZE);
+            for (size_t j = 0; j < gameState.ARENA_SIZE; j++) {
+                gameMap[i].push_back(Node());
```

Comment:
If this is how you create the Node you should give the Node struct a constructor rather than just declaring all of the variables. For example:
```
struct Date {
    int day;
    int month;
    int year;

    Date():day(0),
           month(0),
           year(0){}
};
```
Otherwise use the { a, b, c, ... } structure.